codice/ddf
# DDF-2162 Add the ability to provide attribute mixins for metacard types

#### What does this PR do?
- Adds an API for registering 'global' attributes that will be mixed/injected into all metacard types
- Augments the JSON metacard type definition functionality by allowing attribute mixins to be defined in JSON files
- Updates the metacard type registry to add the mixed-in attributes to the returned metacard type
- Updates the GeoJSON and XML input transformers to fetch metacard types from the metacard type registry so they will have the correct mixins
- Updates the standard catalog framework implementation to mix in the attributes on creates and updates

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@rzwiefel 
@bdeining 

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@jlcsmith
@kcwire 
@pklinef

#### How should this be tested?
Define some new attributes and metacard types using the JSON definition feature and make your new attributes mixins. Ingest some GeoJSON or XML metacards with values for the new attributes you defined. Verify that these attributes are present on each ingested metacard, even metacards of type `ddf.metacard`.

#### What are the relevant tickets?
[DDF-2162](https://codice.atlassian.net/browse/DDF-2162)

#### Checklist:

### ====Notes====
This is an early-stage PR. It lacks unit tests, integration tests, and documentation. Those will be implemented after the core feature and API changes have stabilized.
